src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -700,6 +734,33 @@ public Bindings(TextLoader parent, Column[] cols, IMultiStreamSource headerFile,
                     else if (headerFile == null && parent.HasHeader)
                         Cursor.GetSomeLines(dataSample, 1, ref lines);
 
+                    if (needColumnLocalization)
+                    {
+                        if (lines == null)
+                            throw ch.ExceptUserArg(nameof(dataSample), "Can't determinate column index without sample data");
+
+                        var header = Parser.GetHeaderFields(parent, lines[0]);
+                        var headerMapping = Enumerable.Zip(header.Indices, header.Spans,
+                            (index, span) => new
+                            {
+                                Index = index,
+                                Span = span
+                            })
+                            .Where(x => !x.Span.IsEmpty)
+                            .GroupBy(x => x.Span.ToString().ToLowerInvariant());
```

Comment:
>ToLowerInvariant [](start = 60, length = 16)

Column names are usually case sensitive, so there is no need to call `ToLowerInvariant`.